File: 프로그래머스/JavaScript/0단계/배열 두 배 만들기.js
// 정수 배열 numbers가 매개변수로 주어집니다. numbers의 각 원소에 두배한 원소를 가진 배열을 return하도록 solution 함수를 완성해주세요.


// 내 풀이
function solution(numbers) {
    var answer = [];
    for(let i = 0; i < numbers.length; i ++){
     answer.push(numbers[i]*2)   
    }
    return answer;
}

// 다른사람 풀이(reduce 사용법)
function solution(numbers) {
    return numbers.reduce((a, b) => [...a, b * 2], []);
}

// 다른사람 풀이(map 사용법)
const solution = (numbers) => numbers.map((number) => number * 2)

File: 프로그래머스/Python/0단계/배열 두 배 만들기.py
# 정수 배열 numbers가 매개변수로 주어집니다. 
# numbers의 각 원소에 두배한 원소를 가진 배열을 return하도록 solution 함수를 완성하는 문제.


# 내 풀이

def solution(numbers):
    answer = []
    for i in numbers:
        answer.append(i*2)
    return answer
  
  
# 다른사람 풀이1

def solution(numbers):
    return [num*2 for num in numbers]  # 리스트 컴프리헨션: [표현식 for 항목 in 반복가능객체 if 조건문] 
  
# 다른사람 풀이2

def solution(numbers):
    return list(map(lambda x: x * 2, numbers))
